.evergreen/config_generator/components/cse/darwinssl.py

```diff
@@ -14,15 +14,21 @@
 # pylint: disable=line-too-long
 # fmt: off
 COMPILE_MATRIX = [
-    ('macos-1100', 'clang', None, ['cyrus']),
+    ('macos-14', 'clang', None, ['cyrus']),
+
+    ('macos-11-arm64', 'clang', None, ['cyrus']),
+    ('macos-14-arm64', 'clang', None, ['cyrus']),
 ]
 
 # TODO (CDRIVER-3789): test cse with the 'sharded' topology.
 TEST_MATRIX = [
-    ('macos-1100', 'clang', None, 'cyrus', ['auth'], ['server', 'replica' ], ['4.2', '4.4', '5.0', '6.0'                ]),
+    ('macos-14', 'clang', None, 'cyrus', ['auth'], ['server', 'replica' ], ['4.2', '4.4', '5.0']),
+
+    ('macos-11-arm64', 'clang', None, 'cyrus', ['auth'], ['server', 'replica' ], ['6.0']),
+    ('macos-14-arm64', 'clang', None, 'cyrus', ['auth'], ['server', 'replica' ], ['6.0']),
```

Comment:
```suggestion
    ('macos-14', 'clang', None, 'cyrus', ['auth'], ['server'], ['4.2', '4.4', '5.0']),

    ('macos-11-arm64', 'clang', None, 'cyrus', ['auth'], ['server'], ['6.0']),
    ('macos-14-arm64', 'clang', None, 'cyrus', ['auth'], ['server'], ['6.0']),
```

Suggest removing `replica` for `<7.0` server tests to reduce macOS testing. The openssl and winssl CSE tasks similarly test only `server` for `<7.0` server. I expect testing only `server` for `<7.0` was originally intended (given the comment `# Test 7.0+ with a replica set`).